tests/src/core/testqgsexpression.cpp

```diff
@@ -2443,6 +2443,43 @@ class TestQgsExpression: public QObject
       QCOMPARE( badArray.evalErrorString(), QString( "Cannot convert 'not an array' to array" ) );
     }
 
+    void compare_arrays()
+    {
+      QCOMPARE( QgsExpression( "array() = array()" ).evaluate(), QVariant( true ) );
+      QCOMPARE( QgsExpression( "array(NULL) = array(NULL)" ).evaluate(), QVariant( true ) );
```

Comment:
I agree, but regarding PostgreSQL ( Array[NULL] = Array[NULL] ) => true.
For me this make sense, for instance if you want to make a group_by on two columns.
Array[1, NULL] = Array[1, NULL] => True.
Without this you will lost some records.